packager/packager.cc

```diff
@@ -438,12 +430,16 @@ bool CreateRemuxJobs(const StreamDescriptorList& stream_descriptors,
       if (hls_playlist_name.empty())
         hls_playlist_name = base::StringPrintf("stream_%d.m3u8", stream_number);
 
-      muxer_listener.reset(new HlsNotifyMuxerListener(hls_playlist_name, name,
+      muxer_listeners.emplace_back(new HlsNotifyMuxerListener(hls_playlist_name, name,
                                                       group_id, hls_notifier));
     }
 
-    if (muxer_listener)
-      muxer->SetMuxerListener(std::move(muxer_listener));
+    if (!muxer_listeners.empty()) {
+        std::unique_ptr<MuxerListener> muxer_listener;
+        std::unique_ptr<CombinedMuxerListener> combined_muxer_listener(new CombinedMuxerListener(&muxer_listeners));
+        muxer_listener = std::move(combined_muxer_listener);
+        muxer->SetMuxerListener(std::move(muxer_listener));
```

Comment:
You can do the below instead:

```C++
muxer->SetMuxerListener(std::move(combined_muxer_listener));
```

The intermediate variable muxer_listener is not needed.